hooks/useFetch.js

```diff
@@ -108,18 +104,27 @@ const useFetch = (url, refresh = false) => {
   const [loading, setLoading] = useState(true)
 
   useEffect(() => {
-    const { loading, data, refresh: startRefresh } = fetchData(url)
-    const sub = loading.subscribe(setLoading).add(
-      data.subscribe((ev) => {
-        setData(ev.value.body)
-        setError(ev.error)
+    const fetcher = fetchData(url).pipe(
+      tap((ev) => {
+        // maybe check for no errors? shrug
+        if (ev.hasValue) {
+          setLoading(false)
+        }
       }),
     )
 
-    if (refresh) {
-      sub.add(startRefresh.subscribe())
+    let refresher = fetcher
+    if (!refresh) {
+      refresher = fetcher.pipe(take(1))
     }
 
+    const sub = refresher.subscribe((ev) => {
+      if (ev.error) {
```

Comment:
Is there a reason you're guarding the error here? My thinking was that if `ev.error` is undefined, on the next refresh it'll get unset (imagine retries).

Reply:
Notice the `return` in the subsequent line `return setError(ev.error)` which will avoid a null pointer exception for `ev.value.body` when an error occurs because `ev.value` will be null. Maybe not the most self-explanatory way to avoid this execption.